components/callbacks/fx.go

```diff
@@ -48,9 +50,14 @@ var Module = fx.Module(
 
 func HTTPCallerProviderProvider(
 	clusterMetadata cluster.Metadata,
-	httpClientCache *cluster.FrontendHTTPClientCache,
+	rpcFactory common.RPCFactory,
 	logger log.Logger,
-) HTTPCallerProvider {
+) (HTTPCallerProvider, error) {
+	internalClient, err := rpcFactory.CreateLocalFrontendHTTPClient()
```

Comment:
I think this client is only for requests within the same cluster, right? If so, we will need a mechanism for getting remote cluster frontend clients to auto-resolve callbacks targeting a different cluster, similar to how we are using `rpc.CreateRemoteFrontendGRPCConnection` and `clientfactory.NewRemoteFrontendClientWithTimeout`

Reply:
Hmm... I was counting on forwarding to do this for us but I guess we may get request from a peer cluster that has a namespace that we don't know about.

I'll change this. Good point!